### PR TITLE
Add Ignore external dependencies flag

### PR DESCRIPTION
**Pull request by gpiper14**
Adds a `--terragrunt-ignore-external-dependencies` argument that will not prompt the user if they want to include an external dependency.  This is useful if you just need to run a targeted subset of modules of a larger config

### CODE REVIEW

File: options/options.go [hunk 1]
```diff
@@ -117,28 +120,29 @@ func NewTerragruntOptions(terragruntConfigPath string) (*TerragruntOptions, erro
...
+		NonInteractive:             false,
+		TerraformCliArgs:           []string{},
+		WorkingDir:                 workingDir,
+		Logger:                     logger,
+		Env:                        map[string]string{},
+		Source:                     "",
+		SourceUpdate:               false,
+		DownloadDir:                downloadDir,
+		IgnoreDependencyErrors:     false,
+		IgnoreExternalDependencies: false,
```
Comment (brikis98) [inline options/options.go]: There's a `Clone` method below that needs to be updated too

**Comment by gpiper14** [comment]
> Thanks for the PR!
> 
> Could you add a test for this new flag?
@brikis98 
Yeah sure.  I'll try to knock that out this week.  I'm assuming an integration tests?

**Comment by brikis98** [comment]
> Yeah sure. I'll try to knock that out this week. I'm assuming an integration tests?

Yes please! See `test/integration_test.go`. Thx 👍

**Comment by gpiper14** [comment]
@brikis98 .  Added some test. lmk if i need to fix anything.  fairly new to go

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terragrunt/releases/tag/v0.18.3

**Pull request merged**